hs_geographic_feature_resource/receivers.py

```diff
@@ -212,51 +212,65 @@ def geofeature_pre_create_resource(sender, **kwargs):
     tmp_dir = None
     try:
         files = kwargs['files']
-        metadata = kwargs['metadata']
-        validate_files_dict = kwargs['validate_files']
         fed_res_fnames = kwargs['fed_res_file_names']
         fed_res_path = kwargs['fed_res_path']
-        file_selected = False
-
-        if files:
-            file_selected = True
-            file_list = [file.name for file in files]
-        elif fed_res_fnames:
-            file_list = utils.get_fed_zone_files(fed_res_fnames)
-            if file_list:
-                file_selected = True
-
-        if file_selected:
-            files_type = check_uploaded_files_type(files, file_list)
+        metadata = kwargs['metadata']
+        validate_files_dict = kwargs['validate_files']
+
+        if files and fed_res_fnames:
+            validate_files_dict['are_files_valid'] = False
+            validate_files_dict['message'] = 'Please upload files from either local disk or irods, not both.'
+            return
+
+        if files or fed_res_fnames:
+            files_list = [f for f in files]
```

Comment:
@hyi I was trying to use the same interface to handle both files uploaded from local and irods to avoid changing existing code too much. The files from fed's irods are already downloaded onto django server (in order to extract metadata), then I use UploadedFile() to warp them and pass those UploadedFile objs to existing code. I know if the file is a remote file, using UploadedFile () to wrap them may cause performance issue, but in this case they are already local to django server.